editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search.tsx

```diff
@@ -1,132 +1,283 @@
 import Input from "ui/shared/Input";
 
-const mockData: SearchResult[] = [
-  {
-    nodeId: "abc123",
-    nodeType: ComponentType.Question,
-    nodeTitle: "Is the property in Lambeth?",
-    text: "Lambeth example biodiversity text",
-    path: ["_root", "xyz123", "abc123"],
-  },
-  {
-    nodeId: "abc456",
-    nodeType: ComponentType.Notice,
-    nodeTitle: "It looks like the property is not in Lambeth",
-    text: "Lambeth example biodiversity text",
-    path: ["_root", "xyz123", "abc123"],
-  },
-  {
-    nodeId: "abc789",
-    nodeType: ComponentType.Question,
-    nodeTitle: "What are you applying about?",
-    text: "Lambeth example biodiversity text",
-    path: ["_root", "xyz123", "abc123"],
-  },
-];
-
-export interface SearchResult {
-  nodeId: string;
-  nodeType: ComponentType;
-  nodeTitle?: string;
+const SearchResultRoot = styled(List)(({ theme }) => ({
+  width: "100%",
+  gap: theme.spacing(2),
+  display: "flex",
+  flexDirection: "column",
+}));
+
+const SearchResultCardRoot = styled(ListItemButton)(({ theme }) => ({
+  padding: theme.spacing(1),
+  border: `1px solid ${theme.palette.common.black}`,
+  display: "block"
+}));
+
+const ExternalPortalCard = styled(ListItemButton)(({ theme }) => ({
+  backgroundColor: "black",
+  color: theme.palette.common.white,
+  "&:hover": {
+    backgroundColor: "black",
+    "& p:last-of-type": {
+      textDecoration: "underline",
+    }
+  }
+}));
+
+const SearchResults: React.FC<{ results: SearchResults<IndexedNode> }> = ({
+  results,
+}) => {
+  return (
+    <>
+      <Typography variant="h3" mb={1}>
+        {!results.length && "No matches found"}
+        {results.length === 1 && "1 result:"}
+        {results.length > 1 && `${results.length} results:`}
+      </Typography>
+
+      <SearchResultRoot>
+        {results.map((result) => (
+          <ListItem key={result.item.id} disablePadding>
+            <SearchResultCard result={result} />
+          </ListItem>
+        ))}
+      </SearchResultRoot>
+    </>
+  );
+};
+
+interface HeadlineProps {
   text: string;
-  path: string[];
+  matchIndices: [number, number][];
+  variant: "data";
 }
 
-const SearchResults: React.FC<{ results: SearchResult[] }> = ({ results }) => {
+const Headline: React.FC<HeadlineProps> = ({ text, matchIndices, variant }) => {
+  const isHighlighted = (index: number) =>
+    matchIndices.some(([start, end]) => index >= start && index <= end);
+
   return (
-    <Box
-      sx={{ width: "100%", gap: 2, display: "flex", flexDirection: "column" }}
-    >
-      {results.map((result) => (
-        <SearchResultCard key={result.nodeId} {...result} />
+    <>
+      {text.split("").map((char, index) => (
+        <Typography
+          fontWeight={isHighlighted(index) ? FONT_WEIGHT_BOLD : "regular"}
+          component="span"
+          p={0}
+          key={`headline-character-${index}`}
+          fontFamily={
+            variant === "data" ? '"Source Code Pro", monospace' : "inherit"
```

Comment:
nit / future thing: it'd be handy to have our "data" font set in MUI's Theme and accessible via `theme.text.code` or something similar ! Feels like a nice style pattern we're repeating more & more lately

Reply:
Very very good shout! Picked this up here, and will add to this PR once merged https://github.com/theopensystemslab/planx-new/pull/3474